drivers/src/hand_off.rs

```diff
@@ -89,18 +89,29 @@ pub enum DataVaultRegister {
 impl TryInto<DataStore> for HandOffDataHandle {
     type Error = CaliptraError;
     fn try_into(self) -> Result<DataStore, Self::Error> {
-        let vault = Vault::try_from(self.vault())
-            .unwrap_or_else(|_| report_handoff_error_and_halt("Invalid Vault", 0xbadbad));
+        let vault = Vault::try_from(self.vault()).unwrap_or_else(|_| {
+            report_handoff_error_and_halt(
+                "Invalid Vault",
+                CaliptraError::DRIVER_HANDOFF_INVALID_VAULT.into(),
+            )
+        });
         match vault {
             Vault::KeyVault => Ok(DataStore::KeyVaultSlot(
-                KeyId::try_from(self.reg_num() as u8)
-                    .unwrap_or_else(|_| report_handoff_error_and_halt("Invalid KeyId", 0xbadbad)),
+                KeyId::try_from(self.reg_num() as u8).unwrap_or_else(|_| {
+                    report_handoff_error_and_halt(
+                        "Invalid KeyId",
+                        CaliptraError::DRIVER_HANDOFF_INVALID_KEY_ID.into(),
+                    )
+                }),
             )),
             Vault::DataVault => match self.reg_type() {
                 1 => {
                     let entry = DataStore::DataVaultSticky4(
                         ColdResetEntry4::try_from(self.reg_num() as u8).unwrap_or_else(|_| {
-                            report_handoff_error_and_halt("Invalid ColdResetEntry4", 0xbadbad)
+                            report_handoff_error_and_halt(
+                                "Invalid ColdResetEntry4",
+                                CaliptraError::DRIVER_HANDOFF_INVALID_COLD_RESET_ENTRY4.into(),
```

Comment:
Why are we halting as opposed to returning Err(CaliptraError::DRIVER_HANDOFF_INVALID_COLD_RESET_ENTRY4)?

Reply:
(doesn't need to be fixed in this PR, as this isn't new behavior, but I'm curious what is going on here)

Reply:
Will need @rusty1968 to respond to this.